lib/ansible/parsing/yaml/objects.py

```diff
@@ -113,6 +113,9 @@ def __init__(self, ciphertext):
         self.vault = None
         self._ciphertext = to_bytes(ciphertext)
 
+    __class__ = text_type
+
```

Comment:
shouldn't we just inherit from AnsibleUnicode instead of AnsibleBaseYAMLObject

Reply:
Yeah, this looks dangerous. But also inheriting from string is rather problematic. I recall attempting this a year ago in our migration script and it was rather painful.

Reply:
that is smarter :) agreed